tables/tests/test_expression.py

```diff
@@ -149,6 +149,25 @@ def test01_out(self):
         self.assertTrue(common.areArraysEqual(r1, r2),
                         "Evaluate is returning a wrong value.")
 
+    def test02_out(self):
+        """Checking that expression is correctly evaluated when slice is
+        outside of data samples (`out` param)"""
+        expr = tables.Expr(self.expr, self.vars)
+        # maybe it's better to use the leading dimemsion instead?
+        maxshape = max(self.shape)
+        start, stop, step = (maxshape + 1, maxshape + 2, None)
```

Comment:
This test is a good idea for checking outside ranges, but you are missing the current issue, when start and stop are identical. Maybe you could test it when start == stop too.